src/ansys/openapi/common/_oidc.py

```diff
@@ -101,54 +91,58 @@ def __init__(
             if "scope" in self._authenticate_parameters
             else []
         )
-        self._oauth_session = OAuth2Session(
+
+        self._auth = OAuth2AuthorizationCodePKCE(
+            authorization_url=self._well_known_parameters["authorization_endpoint"],
+            token_url=self._well_known_parameters["token_endpoint"],
+            redirect_uri_port=32284,
+            audience=self._authenticate_parameters["apiAudience"] if "apiAudience" in self._authenticate_parameters else None,
             client_id=self._authenticate_parameters["clientid"],
-            redirect_uri=self._authenticate_parameters["redirecturi"],
             scope=scopes,
+            session=self._initial_session
         )
-        set_session_kwargs(self._oauth_session, self._api_session_configuration)
 
-        if "offline_access" in scopes:
-            self._configure_token_refresh()
+        self._auth.refresh_data.pop('audience', None)
```

Comment:
Is this line needed? It looks like you are setting the audience in the constructor above.